cmd/db_repair/repair.go

```diff
@@ -85,46 +38,28 @@ func (pc *PostgresClient) RepairEvents(cfClient cf.Client) (err error) {
 	if err := rows.Err(); err != nil {
 		return err
 	}
-
-	// cache all the things
-	orgs, err := cfClient.GetOrgs()
-	if err != nil {
-		return err
-	}
-	spaces, err := cfClient.GetSpaces()
-	if err != nil {
-		return err
-	}
-	services, err := cfClient.GetServices()
-	if err != nil {
-		return err
-	}
-	servicePlans, err := cfClient.GetServicePlans()
-	if err != nil {
-		return err
-	}
-
-	// fetch all the running apps
 	apps, err := cfClient.GetApps()
 	if err != nil {
 		return err
 	}
+	fmt.Println("apps:", len(apps))
+	var fakeEventsInserted int
 	for _, app := range apps {
-		// Skip non running apps (FIXME: ListApps() could probably filter this)
-		if app.State != StateStarted {
+		// Skip non running apps
+		if app.State != db.StateStarted {
 			continue
 		}
+
+		// Skip if we already have events for this app
+		if seen := knownGuids[app.Guid]; seen {
+			continue
+		}
+
 		// Skip anything updated since we started collecting events
 		updated, err := time.Parse(time.RFC3339, app.UpdatedAt)
 		if err != nil {
 			return err
 		}
-		fmt.Println("app", app.Guid)
-		// Skip if we already have events for this app
-		if seen := knownGuids[app.Guid]; seen {
-			continue
-		}
-		fmt.Println("detected missing app", app.Guid)
 		if updated.After(*firstEventTimestamp) {
```

Comment:
This is about not creating fake events for apps that have already or are about to emit STARTED events on their own.

It's possible that checking the "created at" would be more appropriate, but updated seemed safer to me, since creating an bad fake event (and charging more) is worse than skipping the event (and undercharging).